jpa/org.eclipse.persistence.jpa.jpql/src/main/java/org/eclipse/persistence/jpa/jpql/AbstractGrammarValidator.java

```diff
@@ -1791,35 +1792,6 @@ protected final boolean isNewerThanOrEqual(JPAVersion version) {
         return getJPAVersion().isNewerThanOrEqual(version);
     }
 
-    /**
-     * Determines whether the given sequence of characters is a numeric literal or not. There are
-     * two types of numeric literal that is supported:
-     * <ul>
-     *    <li>Decimal literal</li>
-     *    <li>Hexadecimal literal</li>
-     * </ul>
-     *
-     * @param text The sequence of characters to validate
-     * @return <code>true</code> if the given sequence of characters is a valid numeric literal;
-     * <code>false</code> otherwise
-     */
-    protected boolean isNumericLiteral(String text) {
```

Comment:
I didn't delete this method, just moved it to the NumericLiteral class and renamed to `hasValidValue`, so that it can be used also in the parser, not only in the validator.

Reply:
I found a different way to fix the bug, so I reverted removing this method.